File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        x = 0
        sum_digits = 0
        multiplier = 1
        
        while n > 0:
            digit = n % 10
            if digit > 0:
                x += digit * multiplier
                sum_digits += digit
                multiplier *= 10
            n //= 10
            
        return x * sum_digits

File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {
        long long x = 0;
        long long multiplier = 1;
        long long sum = 0;

        while(n > 0){
            int temp = n % 10;
            if(temp > 0){
                x += (temp*multiplier);
                sum += temp;
                multiplier *= 10;
            }
            n /= 10;
        }
        return x * sum;
    }
};